packages/fiori-elements-writer/src/index.ts

```diff
@@ -53,7 +53,12 @@ async function generate<T>(basePath: string, data: FioriElementsApp<T>, fs?: Edi
     const rootTemplatesPath = join(__dirname, '..', 'templates');
     // Add templates common to all template types
     // Common files
-    const ignore = [feApp.appOptions?.typescript ? '**/*.js' : '**/*.ts'];
+    const isV2FETypesAvailable = feApp.ui5?.version
```

Comment:
What does V2 refer to here please? There is no check for odata version. 

Also the same coercion of ui5 version is being done multiple times and could be consolidated. Probably this type script  specific code should be moved to another function so it can be tested in isolation. I think all this TS related code should only be done if TS is requested but I dont se that check being done either?

Reply:
Add comment regarding isV2FETypesAvailable and add check for typescript flag before executing the logic.